Lib/importlib/metadata.py

```diff
@@ -48,7 +48,7 @@ def name(self):
 
 
 class EntryPoint(
-        collections.namedtuple('EntryPointBase', 'name value group')):
+        collections.namedtuple('EntryPointBase', 'name value group dist')):
```

Comment:
I'm a little uneasy exposing the distribution name as 'dist' and the distribution as 'distribution', especially with `dist` is used in the same code to refer to the Distribution object. I'd prefer if we can find an implementation that avoids this dual-mode of representing the Distribution. Could the EntryPoint instead just always keep a reference to the Distribution?

Reply:
Sure, and `pkg_resources` does just that.(if we're trying to match the implentation)

The reason why I didn't follow that path is because, at least so far, no one was missing it.

Now, without iterating thought all entry-points, how do you know the name of then distribution to load the instance? We need a "simple" object to allow EntryPoint to be pickleable. Unless, `Distribution` is also pickleable, I'd need to confirm.

Would `dist_name` be better than `dist`?

Reply:
I think we found a way to avoid storing the dist and name separately.